### PR TITLE
adds react router, redux and tpescript support

### PR DESCRIPTION
**Pull request by shubhamnarang11**

### CODE REVIEW

**Comment by sumantasau** [comment]
Its good project structure. Want to know usage of modal, reducer and action folder structure

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,12 @@
-import logo from './logo.svg';
+//import logo from './logo.svg';
```
Comment (shubhamnarang11) [inline src/App.tsx]: @sumantasau Its not good practice to push commented code.

File: src/actions/Login.action.ts [hunk 2]
```diff
@@ -0,0 +1 @@
+export {};
```
Comment (shubhamnarang11) [inline src/actions/Login.action.ts]: Please install Prettier or any other extension and format the files before pushing.

File: src/components/Profile/Profile.css [hunk 3]
```diff
@@ -0,0 +1,106 @@
+* {
```
Comment (shubhamnarang11) [inline src/components/Profile/Profile.css]: As we have discussed, we will move forward with scss, so you shouldn't mix plain css in our project.

File: src/components/Profile/Profile.css [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+  
+  .left {
+    padding: 20px;
+    float: left;
+    text-align: center;
+    width: 30%; /* The width is 30%, by default */
+  }
+  
+  .profile-image {
+    width: 200px;
```
Comment (shubhamnarang11) [inline src/components/Profile/Profile.css]: These values are fixed in pixels which will not make it responsive for bigger screens or desktops. So, try to change it to % or rem.

File: src/components/Profile/Profile.jsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
+import React, { useState, useRef } from "react";
```
Comment (shubhamnarang11) [inline src/components/Profile/Profile.jsx]: Same for this file, it should be tsx.

File: src/components/Profile/Profile.jsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+   inputFile.current.click();
+  };
+
+  return (
+    <div className="container">
+      <div className="left">
+        <img src={avatar_image} className="profile-image" alt="user profile" onClick={onImageClick} />
+        <img src={camera_image} className="small-camera" alt="small camera" onClick={onImageClick} />
+        
+        <input type='file' id='file' ref={inputFile} style={{display: 'none'}}/>
```
Comment (shubhamnarang11) [inline src/components/Profile/Profile.jsx]: Its not a best practice to use inline styles.

**Pull request merged**